sympy/physics/control/tests/test_lti.py

```diff
@@ -577,53 +574,186 @@ def test_Feedback_construction():
+    raises(ValueError, lambda: TransferFunctionMatrix([TF1, TF2, tf4], (3, 1), s))
+    raises(ValueError, lambda: TransferFunctionMatrix([[TF1], [TF3, tf5]], (2, 2), s))
+    raises(TypeError, lambda: TransferFunctionMatrix([[TF1, TF2], [TF3, Matrix([1, 2])]], (2, 2), s))
+    raises(ValueError, lambda: TransferFunctionMatrix([[TF1, tf4], [TF3, tf5]], (2, 2), s))
+
+
+def test_TransferFunctionMatrix_functions():
+    tf5 = TransferFunction(a1*s**2 + a2*s - a0, s + a0, s)
+    tf6 = TransferFunction(a0*s**3 + a1*s**2 - a2*s, b0*p**4 + b1*p**3 - b2*s*p, s)
+
+    tfm1 = TransferFunctionMatrix([TF1, TF2], (2, 1), s)
+    assert -tfm1 == TransferFunctionMatrix([-TF1, -TF2], (2, 1), s)
+
+    tfm2 = TransferFunctionMatrix([[TF1, -TF2], [TF3, tf5]], (2, 2), s)
+    assert -tfm2 == TransferFunctionMatrix([[-TF1, TF2], [-TF3, -tf5]], (2, 2), s)
+
+    tfm3 = TransferFunctionMatrix([[TF1, TF2, TF3], [tf5, -TF1, -TF3]], (2, 3), s)
+    assert -tfm3 == TransferFunctionMatrix([[-TF1, -TF2, -TF3], [-tf5, TF1, TF3]], (2, 3), s)
+
+    tfm4 = TransferFunctionMatrix([TF1, TF2, TF3], (3, 1), s)
+    assert tfm4.is_proper
+    assert not tfm4.is_strictly_proper
+    assert not tfm4.is_biproper
+
+    tfm5 = TransferFunctionMatrix([[TF1, TF2], [TF3, -tf5], [-tf5, TF2]], (3, 2), s)
+    assert not tfm5.is_proper
+    assert not tfm5.is_strictly_proper
+    assert not tfm5.is_biproper
+
+    tfm6 = TransferFunctionMatrix([[TF1, TF2], [TF3, -TF3]], (2, 2), s)
+    assert tfm6.is_proper
+    assert not tfm6.is_strictly_proper
+    assert not tfm6.is_biproper
+
+    tfm7 = TransferFunctionMatrix([-TF1, -tf6], (2, 1), s)
+    assert not tfm7.is_proper
+    assert not tfm7.is_strictly_proper
+    assert not tfm7.is_biproper
+
+
+def test_TransferFunctionMatrix_addition_and_subtraction():
+    tf = TransferFunction(a0*p, a1*p**2 + a2*p - a0, p)
+    tfm1 = TransferFunctionMatrix([TF1, TF2], (2, 1), s)
+    tfm2 = TransferFunctionMatrix([-TF2, -TF1], (2, 1), s)
+    tfm3 = TransferFunctionMatrix([TF1, TF1], (2, 1), s)
+    tfm4 = TransferFunctionMatrix([tf], (1, 1), p)
+    tfm5 = TransferFunctionMatrix([TF2, TF1, TF3], (3, 1), s)
+    tfm6 = TransferFunctionMatrix([-TF1, TF2, -TF3], (3, 1), s)
+    tfm7 = TransferFunctionMatrix([TF2], (1, 1), s)
+
+    # addition
+    assert tfm1 + tfm2 == Parallel(tfm1, tfm2)
+    assert tfm3 + tfm1 == Parallel(tfm3, tfm1)
+    assert tfm2 + (tfm3 + tfm1) == Parallel(tfm2, tfm3, tfm1)
+    assert tfm1 + tfm3*tfm7 == Parallel(tfm1, Series(tfm3, tfm7))
+
+    c = symbols("c", commutative=False)
+    raises(ValueError, lambda: tfm1 + Matrix([1, 2, 3]))
+    raises(ValueError, lambda: tfm2 + c)
+    raises(ValueError, lambda: tfm3 + tfm4)
+    raises(ValueError, lambda: tfm1 + (s - 1))
+    raises(ValueError, lambda: tfm1 + 8)
+    raises(ValueError, lambda: (1 - p**3) + tfm1)
+    raises(ValueError, lambda: tfm1 + (TF1 + TF2))
+    raises(ShapeError, lambda: tfm2 + (tfm5 + tfm6))
```

Comment:
It would be helpful to write a comment for each of these explaining what they are testing as it isn't apparent without searching for the definitions of the transfer functions.